apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1307,6 +1280,7 @@ NetSimRouterNode.prototype.generateDnsResponse_ = function (message, onComplete)
       this.shard_,
       autoDnsNodeID,
       routerNodeID,
+      message.simulatedBy,
```

Comment:
Picking a simulating node #2:
Auto-DNS responses are always simulated by the same node that was simulating the DNS request.